### DISCUSSION TITLE
Best approach for processing rds result set (denormalized dataset) into hierarchical dataclass

### DISCUSSION THREAD

**Discussion by MartyBolton** [bmsan/DictGest#30] [Q&A] [answered]
Hi, do you have example of taking a flat denormalized SQL dataset, i.e. list of rows but then converting that to a hierarchy of dataclasses with list of objects?  Is it as simple as annotating the fields which denote the keys or sub-classes?  
For example
Order-Detail Id, Customer Name, Cost,  Order Id, Product Id, Product Name, 
1, Mary, $20, 123, 1, Soda, 
2, Joe, $21, 123, 1, Soda
3, Joe, $21, 123, 2, Chips

into...
Order-Detail Table
Order Table
Customer Table
Product Table

data object
thank you!

**Comment by bmsan** [comment]
Would the following work for you?

```py
keys = {Order-Detail Id, Customer Name, Cost, Order Id, Product Id, Product Name}
data = [[1, Mary, $20, 123, 1, Soda],
[2, Joe, $21, 123, 1, Soda],
[3, Joe, $21, 123, 2, Chips]]

# Option1
dict_of_lists = {key: [row[col_idx] for for row in data] for col_idx, key in enumerate(keys)}
from_dict(...dict_of_lists...)

# Option2 here you'd have a list of dictionaries, each dictionary will contain a row, each row will be convertible to a dataclass - but not exactly what you asked
rows_as_dict = [{k:v for k, v in zip(keys, row)} for for row in data]
[from_dict(... row...) for row in rows_as_dict]
```

It would probably help in the next release for me to add a from_list with the two behaviours mentioned above.
Thanks for your feedback!

Created https://github.com/bmsan/DictGest/issues/31 for implementing this

**Comment by bmsan** [comment, marked as answer]
@MartyBolton I've also implemented this inside the library. 
You can take a look at the examples in the readme https://github.com/bmsan/DictGest#example-8-populating-from-a-2d-table

The new feature is currently only on the main branch, I will upload a new version to pypi in the following days.
